lib/uploader/uploader.ts

```diff
@@ -491,8 +492,16 @@ export class Uploader {
 			if (!disabledChunkUpload) {
 				logger.debug('Initializing chunked upload', { file, upload })
 
+				let chunksWorkspace: string
+				if (this._isPublic) {
+					const shareToken = root!.split('/').at(-1)!
+					chunksWorkspace = `${getBaseUrl()}/public.php/dav/uploads/${shareToken}`
+				} else {
+					chunksWorkspace = generateRemoteUrl(`dav/uploads/${getCurrentUser()?.uid}`)
+				}
+
 				// Let's initialize a chunk upload
-				const tempUrl = await initChunkWorkspace(encodedDestinationFile, retries)
+				const tempUrl = await initChunkWorkspace(chunksWorkspace, encodedDestinationFile, retries)
```

Comment:
So meaning instead we should just pass the mode here:

```suggestion
				const tempUrl = await initChunkWorkspace(encodedDestinationFile, retries, this._isPublic)
```